### ISSUE TITLE
Yarn project building fails with NPE on yarnInstallDirectoryPath

### ISSUE THREAD

**Issue by nitzan-n** [siouan/frontend-gradle-plugin#117]
Hi,

**Description**

When building a Yarn project, installFrontend fails with NPE (stacktrace snippet below).
Tracing the problem,  in FrontendGradlePlugin, yarnInstallDirectoryPath is not initialized because its initializations are conditioned on task.getYarnEnabled().get(), which is apparently never true at the time of these initializations. 
Removing these conditions from around all the initializations of yarnInstallDirectoryPath fixes the problem and the build succeeds.

Stacktrace:

Caused by: java.lang.NullPointerException
        at org.siouan.frontendgradleplugin.domain.usecase.ResolveExecutionSettings.execute(ResolveExecutionSettings.java:100)
        at org.siouan.frontendgradleplugin.infrastructure.gradle.adapter.GradleScriptRunnerAdapter.execute(GradleScriptRunnerAdapter.java:35)
        at org.siouan.frontendgradleplugin.infrastructure.gradle.AbstractRunCommandTask.execute(AbstractRunCommandTask.java:92)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)

**Environment**
 
- OS: Windows 10 
- JDK: 8 64 bits
- Gradle: Gradle 6.3, Gradle 6.4
- Frontend Gradle plugin: latest pulled from GitHub, 2.0.0, 2.1.0

**Comment by v1nc3n4** [comment]
Hi @nitzan-n,

Thank you for your message.
Actually, I don't see why the `yarnInstallDirectoryPath` variable would be `null` at this time: the `installFrontend` task is being executed, which means it has been configured by Gradle, i.e. the `yarnEnabled` property, if `true`, leads to the initialization of this variable in the `FrontendGradlePlugin` class.
I wasn't able to reproduce the issue with Gradle 5.1 (minimum supported version) and Gradle 6.4.

- Can you provide the exact configuration of the plugin in your test case?
- Does the issue occurs every time, or occasionally?
- If you tried with release 2.1.0, can you add a functional test in the `InstallTaskFuncTest` class that reproduces the issue?
- If not possible, can you provide an archive of the project where the issue occurs regularly?

Thanks in advance for your help.
BR,
Vincent

**Comment by v1nc3n4** [comment]
I forgot one thing: can you provide the Gradle's output when executing the task with debug level (`-d` flag)?

**Comment by nitzan-n** [comment]
Hi,

Thanks for your work on the project and your quick response.

I managed to re-create the issue with the attached gradle.build file.
Specifically, there seems to be some interaction with another plugin in the project, 'nebula.clojure', and the 'org.siouan.frontend' plugin:

- When 'nebula.clojure' is activated (setting line 16 to: apply plugin: 'nebula.clojure') the conditional initialization of the Yarn install path doesn't happen, and all the Frontent build task fail with NPE (see attached file).
- When 'nebula.clojure' is not activated (setting line 16 to: apply plugin: 'java') the conditional initialization of the Yarn install path happens, and the build tasks succeed.

At this point I don't know which plugin is 'to blame' for this, or if there are other plugins that may cause this interaction with 'org.siouan.frontend' when they are also applied with it in the same project.

However, as I suggested, removing the conditions from around all the initializations of Yarn install directory in class FrontendGradlePlugin  
(the lines: task.getYarnInstallDirectory().set(extension.getYarnInstallDirectory()))
solves this issue. 
As far as I can tell, there is no drawback to doing so. So at the very least, it should improve "isolation"(?) between the 'frontend' plugin and other project plugins with no cost.

Thanks.

[build.txt](https://github.com/siouan/frontend-gradle-plugin/files/4638122/build.txt)
[build.gradle.txt](https://github.com/siouan/frontend-gradle-plugin/files/4638123/build.gradle.txt)

**Comment by v1nc3n4** [comment]
Hi,
Thank you for this great analysis! I will continue the job to understand why both plugins cannot live together, using the build definition you provided.
I also agree with the fix you suggested, and I already started to remove the conditional initialization of the `yarnInstallDirectory` variable. Initially, this condition was necessary due to how Gradle was validating task I/O. But I found a way to handle things differently, and it seems to be simpler now.
I will provide the fix in release 3.0 at the beginning of the week.
Thanks again for your kind help!
BR,
Vincent
